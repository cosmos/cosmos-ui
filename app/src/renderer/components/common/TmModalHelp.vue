<template lang="pug">
modal.tm-modal-help(v-if="active" :close="close")
  div(slot='title') Need Help?
  p Something in the app not working correctly? Tell us what went wrong in our <a href="https://riot.im/app/#/room/#cosmos:matrix.org">developer chat</a> or create an issue on <a href="https://github.com/cosmos/voyager/issues">GitHub</a>.
  p Thanks for improving Cosmos Voyager!
</template>

<script>
import { mapGetters } from "vuex"
import Modal from "common/TmModal"
import { TmListItem } from "@tendermint/ui"
export default {
  name: `tm-modal-help`,
  components: {
    TmListItem,
    Modal
  },
  computed: {
    ...mapGetters([`config`]),
    active() {
      return this.config.modals.help.active
    }
  },
  methods: {
    close() {
      this.$store.commit(`setModalHelp`, false)
    }
  }
}
</script>

<style lang="stylus">
@import '~variables'

.tm-modal.tm-modal-help
  z-index z(modalHelp)
</style>
