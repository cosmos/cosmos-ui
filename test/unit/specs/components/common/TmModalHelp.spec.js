import setup from "../../../helpers/vuex-setup"
import TmModalHelp from "common/TmModalHelp"

describe(`TmModalHelp`, () => {
  let wrapper, store
  let { mount } = setup()

  beforeEach(() => {
    let instance = mount(TmModalHelp)
    wrapper = instance.wrapper
    store = instance.store
    store.commit(`setModalHelp`, true)
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`should hide`, () => {
    wrapper.vm.close()
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
