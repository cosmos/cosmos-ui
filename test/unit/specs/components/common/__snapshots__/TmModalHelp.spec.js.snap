// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmModalHelp has the expected html structure 1`] = `
<div
  class="tm-modal-help tm-modal"
>
  <div
    class="tm-modal-container"
  >
    <header
      class="tm-modal-header"
    >
      <!---->
       
      <div
        class="tm-modal-title"
      >
        <div>
          
    Need Help?
  
        </div>
      </div>
       
      <div
        class="tm-modal-icon tm-modal-close"
      >
        <i
          class="material-icons"
        >
          
          close
        
        </i>
      </div>
    </header>
     
    <main
      class="tm-modal-main"
    >
       
      <p>
        
    Something in the app not working correctly? Tell us what went wrong in our
    
        <a
          href="https://riot.im/app/#/room/#cosmos:matrix.org"
        >
          
      developer chat
    
        </a>
        
    or create an issue on
    
        <a
          href="https://github.com/cosmos/voyager/issues"
        >
          
      GitHub
    
        </a>
        .
  
      </p>
       
      <p>
        Thanks for improving Cosmos Voyager!
      </p>
       
      <footer
        class="tm-modal-footer"
      />
    </main>
  </div>
</div>
`;

exports[`TmModalHelp should hide 1`] = `<!---->`;
